### PR TITLE
3.0: Add DistributionConfiguration setting

### PR DESCRIPTION
**Pull request by lukeseawalker**
This setting allow to distribute the AMI in different regions and to set LaunchPermission

Signed-off-by: Luca Carrogu <carrogu@amazon.com>

**Please See** [Git Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions)

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cli/src/pcluster/schemas/imagebuilder_schema.py [hunk 1]
```diff
@@ -112,6 +120,18 @@ def validate_component_value(self, data, **kwargs):
             )
 
 
+class DistributionConfigurationSchema(BaseSchema):
+    """Represent the schema of the ImageBuilder distribution configuration."""
+
+    regions = fields.Str()
+    launch_permission = fields.Str()
```
Comment (yuleiwan) [inline cli/src/pcluster/schemas/imagebuilder_schema.py:127]: Minor: add a json syntactic validator like extra_chef_attributes

File: cli/src/pcluster/templates/imagebuilder_stack.py [hunk 2]
```diff
@@ -181,19 +181,33 @@ def __init__(self, scope: core.Construct, construct_id: str, imagebuild: ImageBu
...
+                region=core.Fn.sub("${AWS::Region}"),
+            )
+        ]
+
+        if dev_settings and dev_settings.distribution_configuration and dev_settings.distribution_configuration.regions:
+            for region in set(map(str.strip, dev_settings.distribution_configuration.regions.split(","))):
+                distributions.append(
+                    imagebuilder.CfnDistributionConfiguration.DistributionProperty(
+                        ami_distribution_configuration=ami_distribution_configuration,
+                        region=region,
```
Comment (yuleiwan) [inline cli/src/pcluster/templates/imagebuilder_stack.py:201]: Will it cause an issue if regions contains ${AWS::Region} region? I think we can replace ${AWS::Region} here with region from cli, and add a property **region** in imagebuilder config, like
```
distributions = [            
          imagebuilder.CfnDistributionConfiguration.DistributionProperty(
                ami_distribution_configuration=ami_distribution_configuration,
                region=imagebuild.region,
            )
]

if dev_settings and dev_settings.distribution_configuration and dev_settings.distribution_configuration.regions:
    region_set = set(map(str.strip, dev_settings.distribution_configuration.regions.split(",")))
    if imagebuild.region in region_set:
       region_set.remove(imagebuild.region)
    for region in region_set:
        distributions.append(
            imagebuilder.CfnDistributionConfiguration.DistributionProperty(
                ami_distribution_configuration=ami_distribution_configuration,
                region=region,
           )
     )
```

File: cli/src/pcluster/schemas/imagebuilder_schema.py (hunk 1 above)
Comment (lukeseawalker) [reply cli/src/pcluster/schemas/imagebuilder_schema.py:127]: done

File: cli/src/pcluster/templates/imagebuilder_stack.py (hunk 2 above)
Comment (lukeseawalker) [reply cli/src/pcluster/templates/imagebuilder_stack.py]: done

**Pull request merged**